blocks/market-commentary/market-commentary.js

```diff
@@ -0,0 +1,124 @@
+import { observe, Viewport } from '../../scripts/blocks-utils.js';
+import { fetchPlaceholders } from '../../scripts/aem.js';
+import { callMockCommenrtaryAPI } from '../../scripts/mockapi.js';
+import {
+  div, a, h4, p, span,
+} from '../../scripts/dom-builder.js';
+
+const placeholders = await fetchPlaceholders();
+function allowedCardsCount() {
+  const deviceType = Viewport.getDeviceType();
+  switch (deviceType) {
+    case 'Desktop':
+      return 4;
+    case 'Tablet':
+      return 2;
+    default:
+      return 1;
+  }
+}
+function createMarketCommentaryCard(cardData) {
+  const {
+    articleUrl, titleText, descriptionText, publicationTimeText, footerTimeText,
+  } = cardData;
+  const mainDiv = div(
+    { class: 'card' },
+    a(
+      { href: articleUrl, target: '_blank' },
+      div(
+        { class: 'content' },
+        h4(titleText),
+        p({ class: 'description' }, descriptionText),
+        div(
+          { class: 'info' },
+          p(
+            span(`${placeholders.powerby}`),
+          ),
+          p(
+            span(`${placeholders.publishedon} ${publicationTimeText}`),
+          ),
+        ),
+      ),
+    ),
+    div(
+      { class: 'footer-row' },
+      span({ class: 'footer-circle' }),
+    ),
+    div({ class: 'footer-time' }, footerTimeText),
+  );
+  return mainDiv;
+}
+
+function updateCarouselView(activeDot) {
+  const dotIndex = parseInt(activeDot.dataset.index, 10);
+  const commentaryContainer = activeDot.closest('.market-commentary-container');
+  const dots = commentaryContainer.querySelectorAll('.dot');
+  const currentActiveDot = commentaryContainer.querySelector('.dot.active');
+  if (currentActiveDot && currentActiveDot.dataset.index === activeDot.dataset.index) {
+    return;
+  }
+  const commentaryTrack = commentaryContainer.querySelector('.market-commentary-track');
+  const cards = Array.from(commentaryTrack.children);
+  let moveDistance = dotIndex * cards[0].offsetWidth;
+  if (Viewport.isDesktop() && dotIndex === dots.length - 1) {
+    moveDistance -= ((cards[0].offsetWidth) * 0.9);
+  }
+  commentaryTrack.style.transform = `translateX(-${moveDistance}px)`;
+  dots.forEach((dot) => dot.classList.remove('active'));
+  dots[dotIndex].classList.add('active');
+}
+
+function updateDots(block) {
+  const track = block.querySelector('.market-commentary-track');
+  const dotsContainer = block.querySelector('.dots-container');
+  const cards = track.querySelectorAll('.card');
+  const dotsCont = cards.length - allowedCardsCount() + 1;
```

Comment:
>The total number of cards is 6 on the page[0], by default it shows 5(5.5?)

It would show 4(4.5)
<img width="1726" alt="image" src="https://github.com/aemsites/icicidirect/assets/19485818/cb1728d8-8ec7-4b1e-9cd0-83e5a466bba4">

Reply:
Made a recording to share more details.

https://github.com/aemsites/icicidirect/assets/57093988/14f07e69-8ada-4f33-a161-e163b99a39ea

Reply:
@JiangLong2019 updated as discussed